### ISSUE TITLE
Errors running kpt fn with kpt v0.27.0

### ISSUE THREAD

**Issue by gkowalski-google** [GoogleContainerTools/kpt#678]
I downloaded the [latest kpt binary](https://googlecontainertools.github.io/kpt/installation/binaries/) and I'm running into an issue running a kpt fn. I am running into different errors on different platforms. Notice that I run into an error running the kpt fn on Ubuntu, whereas on gLinux, the error occurs when I try to sink the output.

## GCP Ubuntu
Error while running function.

Command
```
kpt pkg get https://github.com/gkowalski-google/policy-library.git@feature/add-kpt-setter-for-constraint-target ./policy-library
kpt fn source ./policy-library/samples/ |
    kpt fn run --image gcr.io/config-validator/get-policy-bundle:latest -- bundle=forseti-security
```

Error
```
Error: exit status 126
```

## gLinux
Error while piping function results to kpt sink. I have also noticed that this version is very slow on gLinux. For example, running `kpt version` takes over 15 seconds.

Command
```
kpt pkg get https://github.com/gkowalski-google/policy-library.git@feature/add-kpt-setter-for-constraint-target ./policy-library
kpt fn source ./policy-library/samples/ |
    kpt fn run --image gcr.io/config-validator/get-policy-bundle:latest -- bundle=forseti-security |
    kpt fn sink ./policy-library/policies/constraints/
```

Error
```
Error: yaml: line 29: could not find expected ':'
```

**Comment by phanimarupaka** [comment]
@gkowalski-google Can you point me to the source code of config-validator/get-policy-bundle. The issue is due to creating setter for list value here. https://github.com/gkowalski-google/policy-library/blob/master/samples/iam_restrict_gmail_bigquery_dataset.yaml#L25

**Comment by gkowalski-google** [comment]
Hi @phanimarupaka, ok interesting, thanks for catching that! Is there a different way we should be creating a setter for a list value? [Here is the link the source](https://github.com/forseti-security/policy-library/tree/master/bundler/src); [get_policy_bundle.ts](https://github.com/forseti-security/policy-library/blob/master/bundler/src/get_policy_bundle.ts) is the actual function code, but most of the logic is in [common.ts](https://github.com/forseti-security/policy-library/blob/master/bundler/src/common.ts).

**Comment by phanimarupaka** [comment]
@gkowalski-google Here are the instructions to do so. https://googlecontainertools.github.io/kpt/guides/producer/setters/#setting-lists. 
If I use the previous version of Kpt, the setter is not created and so it is not throwing any error, but with latest kpt version the same image is creating the setter but is mangling the yaml. 

@prachirp Are you facing same issue? Any thoughts?

cc @frankfarzan

**Comment by prachirp** [comment]
Yes, I'm facing a similar issue. Maybe source allows sourcing yaml with errors which get caught when running sink.
Commands:
`kpt pkg get https://github.com/progmaticlab/istio.git/galley/pkg/config/analysis/analyzers/testdata/serviceroleservices.yaml .
kpt fn source serviceroleservices.yaml | kpt fn sink output.yaml`

Error:
`Error: yaml: invalid trailing UTF-8 octet`

**Comment by gkowalski-google** [comment]
I have made [this update for the kpt setter](https://github.com/forseti-security/policy-library/pull/348/commits/aaf210f5281d931da25ee1a95aed83f5d1748b9d) to update the kpt setter and move the kpt reference from the element to the list. Everything is now! The issue on gLinux was resolved with that change, and I realized that the error on Ubuntu was becuase I needed to be running these commands with sudo 😞 . Thanks for your help @phanimarupaka!

I'm still seeing really slow performance on gLinux, it's not causing me any issues but let me know if you'd like me to create a separate ticket for this.
```
$ time kpt version
v0.27.0

real    0m15.578s
user    0m0.069s
sys     0m0.032s
```
